src/NuGet.Core/NuGet.Test.Utility/SimpleTestSetup/ProjectFileUtils.cs

```diff
@@ -0,0 +1,110 @@
+            }
+
+            var lastPropGroup = doc.Root.Elements().Last(e => e.Name.LocalName == "PropertyGroup");
+            lastPropGroup.AddAfterSelf(propertyGroup);
+        }
+
+        public static void AddProperty(XDocument doc, string propertyName, string propertyValue)
+        {
+            var lastPropGroup = doc.Root.Elements().Last(e => e.Name.LocalName == "PropertyGroup");
+            lastPropGroup.Add(new XElement(XName.Get(propertyName), propertyValue));
+        }
+
+        public static void AddItem(XDocument doc,
+            string name,
+            string identity,
+            NuGetFramework framework,
+            Dictionary<string, string> properties,
+            Dictionary<string,string> attributes )
+        {
+            AddItem(doc, name, identity,
+                framework?.IsSpecificFramework == true ? framework.GetShortFolderName() : string.Empty, properties, attributes);
+        }
+
+        public static void AddItem(XDocument doc,
+            string name,
+            string identity,
+            string framework,
+            Dictionary<string, string> properties,
+            Dictionary<string,string> attributes )
+        {
+            var ns = doc.Root.GetDefaultNamespace();
+
+            var itemGroup = new XElement(XName.Get("ItemGroup", ns.NamespaceName));
+            var entry = new XElement(XName.Get(name, ns.NamespaceName));
+            entry.Add(new XAttribute(XName.Get("Include"), identity));
+            itemGroup.Add(entry);
+
+            if (!string.IsNullOrEmpty(framework))
+            {
+                itemGroup.Add(new XAttribute(XName.Get("Condition"), $" '$(TargetFramework)' == '{framework}' "));
+            }
+
+            foreach (var attribute in attributes)
+            {
+                var attr = new XAttribute(XName.Get(attribute.Key), attribute.Value);
+                entry.Add(attr);
+            }
+
+            foreach (var pair in properties)
+            {
+                var subItem = new XElement(XName.Get(pair.Key, ns.NamespaceName), pair.Value);
+                entry.Add(subItem);
+            }
+
+            var lastItemGroup = doc.Root.Elements().LastOrDefault(e => e.Name.LocalName == "ItemGroup");
+            if (lastItemGroup == null)
+            {
+                doc.Root.Elements().Last().AddAfterSelf(itemGroup);
+            }
+            else
+            {
+                lastItemGroup.AddAfterSelf(itemGroup);
+            }
+        }
+
+        public static void SetTargetFrameworkForProject(XDocument doc, string targetFrameworkPropertyName, string targetFrameworkValue)
+        {
+            var pgElement = doc.Root.Descendants("PropertyGroup").FirstOrDefault(t => t.Descendants("TargetFramework").Any());
+            if (pgElement == null)
+            {
+                doc.Root.Descendants("PropertyGroup").FirstOrDefault(t => t.Descendants("TargetFrameworks").Any());
+            }
+
+            if (pgElement != null)
```

Comment:
And what is there is no property group that has `TargetFramework` or `TargetFrameworks` tags?

Add a case for `if (pgElement == null)` where a new property group is added with target framework element.